### ISSUE TITLE
Laravel does not delete bank record

### ISSUE THREAD

**Issue by danilodorgam** [yajra/laravel-oci8#318]
### Summary of problem or feature request

The laravel does not show any error but also does not delete.

### Code snippet of problem

Controller
```
 public function destroy($id)
    {
        //
        $projeto = ProjetoModel::findOrFail($id);
        return response()->json($projeto->delete());
    }
```
Model

```
class ProjetoModel extends Model
{
    //
    use SoftDeletes;
    const  DELETED_AT = 'ST_EXCLUIDO';
    const  CREATED_AT = 'ST_CRIADO';
    const  UPDATED_AT = 'ST_ATUALIZADO';
    const NAME_TABLE = 'PROJETO';
    protected $table = 'PROJETO';
    protected $primaryKey = 'ID_PROJETO';
    protected $fillable = ['NM_PROJETO','SG_PROJETO','ID_RESPONSAVEL'];
    protected $guarded = ['ID_PROJETO','ST_EXCLUIDO','ST_CRIADO', 'ST_ATUALIZADO'];
}

```
### System details

- Operating System ubuntu 16.04.2 LTS
- PHP Version 7.1
- Laravel Version 5.4
- Laravel-OCI8 Version last

**Comment by mstaack** [comment]
@danilodorgam what does  `dd($projeto)` give you? do you get a result?

**Comment by danilodorgam** [comment]
@mstaack 


```

ProjetoModel {#247
  #table: "PROJETO"
  #primaryKey: "ID_PROJETO"
  #fillable: array:3 [
    0 => "NM_PROJETO"
    1 => "SG_PROJETO"
    2 => "ID_RESPONSAVEL"
  ]
  #guarded: array:4 [
    0 => "ID_PROJETO"
    1 => "ST_EXCLUIDO"
    2 => "ST_CRIADO"
    3 => "ST_ATUALIZADO"
  ]
  #connection: null
  #keyType: "int"
  +incrementing: true
  #with: []
  #withCount: []
  #perPage: 15
  +exists: true
  +wasRecentlyCreated: false
  #attributes: array:7 [
    "id_projeto" => "2"
    "nm_projeto" => "PF"
    "sg_projeto" => "pf"
    "id_responsavel" => "1"
    "st_criado" => "2017-07-26 19:14:02"
    "st_atualizado" => "2017-07-27 19:14:02"
    "st_excluido" => null
  ]
  #original: array:7 [
    "id_projeto" => "2"
    "nm_projeto" => "PF"
    "sg_projeto" => "pf"
    "id_responsavel" => "1"
    "st_criado" => "2017-07-26 19:14:02"
    "st_atualizado" => "2017-07-27 19:14:02"
    "st_excluido" => null
  ]
  #casts: []
  #dates: []
  #dateFormat: null
  #appends: []
  #events: []
  #observables: []
  #relations: []
  #touches: []
  +timestamps: true
  #hidden: []
  #visible: []
}
```

**Comment by mstaack** [comment]
try looking what sql is produces and on which connection u are making the call..

try
```\DB::listen(function($sql) {
    var_dump($sql);
});```  in appserviceprovider

and then check your sql...

**Comment by mstaack** [comment]
@danilodorgam also keep in mind you are using SoftDelete Trait!

**Comment by danilodorgam** [comment]
Can you show an example of the code?
Yes, but the DELETED_AT field remains null.

I realized that the same happens when I try to update the registry. It does not change.

**Comment by mstaack** [comment]
check your logs...  what return value do you get from the delete call? if none were delete it should be 0...

**Comment by mstaack** [comment]
maybe this is realted to https://github.com/yajra/laravel-oci8/issues/312

**Comment by danilodorgam** [comment]
![27-07-2017 16-44-53](https://user-images.githubusercontent.com/3344991/28689151-f8e6a8cc-72ea-11e7-8028-1c610726a149.png)

**Comment by mstaack** [comment]
give running a manual delete via laravel query builder a try

**Comment by danilodorgam** [comment]
` DB::table('projeto')->where('ID_PROJETO', '=', 3)->delete();`
So physically deleted from the bank

**Comment by TheZeroNine** [comment]
manually deleting as @mstaack noted works fine, but I don't like it, because I have to type the table name in lots of queries instead of using the `protected $table` property on the model.

Neither `Project::findOrFail($id)->delete()` nor `Project::destroy($id)` works.

**Comment by yajra** [comment]
Delete works fine on all my projects. Maybe you have some event / model listeners that cancels the delete command?

**Comment by TheZeroNine** [comment]
I found the solution.. I've overridden the primary key in my model `protected $primaryKey = 'IDNR'` (columnname uppercase). In `Eloquent\Model` the function `setKeysForSaveQuery()` fetches the key which returned `null` in my case. After changing `protected $primaryKey = 'idnr'` (lowercase) both, deleting and updating, works as expected.

However, it's still weird it returns "true" even if I tried to delete a db record `where IDNR = null`. It may be a correct Statement, but only if I pass `null` as value.
